[HDX-550]  RN: capture HTTP request headers

tzurae opened the pull request:
## Goal
Customers can view the header information in the attributes of the span.

## Result
<img width="1459" alt="result" src="https://github.com/hyperdxio/hyperdx-otel-react-native/assets/5051756/850d9d3c-58d2-46fd-a67a-e89557921cbd">

wrn14897 commented on src/instrumentations/xhr.ts:
```diff
@@ -709,6 +709,7 @@ export function instrumentXHR(config: XhrConfig) {
           this.addEventListener('readystatechange', () => {
             if (this.readyState === XMLHttpRequest.HEADERS_RECEIVED) {
               const headers = this.getAllResponseHeaders().toLowerCase();
+              currentSpan.setAttribute('headers', headers);
```
we want to normalize each header and set it separately (https://github.com/open-telemetry/opentelemetry-js/blob/f3aedb76ff00ca82281f8d2bf94a19714487f1d6/experimental/packages/opentelemetry-instrumentation-http/src/utils.ts#L583)

wrn14897 replied:
Also any chance we could extract request headers as well ?

tzurae replied:
Due to security and privacy considerations, browsers do not provide an API to access request headers through XMLHttpRequest (XHR). We can only obtain custom headers set by the user, but we currently do not offer a related API to our developers. Therefore, it may not be possible to implement the retrieval of request headers at this time.

tzurae replied:
> we want to normalize each header and set it separately (https://github.com/open-telemetry/opentelemetry-js/blob/f3aedb76ff00ca82281f8d2bf94a19714487f1d6/experimental/packages/opentelemetry-instrumentation-http/src/utils.ts#L583)

I have made the modifications. Could you please review them? Thank you.

tzurae commented:
Added information on response headers and also included a networkHeadersCapture flag. Additionally, updated README.md. Currently synced with the original repository's upstream commit db5bf0fb7c8516e26d867f359910c76e7e258d37, and there are no breaking changes upon execution.

ernestii commented:
created a new PR without upstream changes here: https://github.com/hyperdxio/hyperdx-otel-react-native/pull/8
